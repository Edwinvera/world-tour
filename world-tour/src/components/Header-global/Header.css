/* @import url('') */
/* Earth logo ⬇ */

.earth-logo {
  position: absolute;
  height: 30vmin;
  top: 20px;
  right: 42vw;
  cursor: pointer;
  animation: earth-logo-spin infinite 20s linear;
}

@keyframes earth-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}

/* Earth logo ⬆ */

.app-header {
  display: flex;
  justify-content: flex-end;
  background-color: #282c34;
  position: relative;
}
li{
  display: inline-block;
  list-style-type: none;
  padding: 15px;
}
ul li:nth-child(n+3){
  border-left: solid 1px grey
}
.menu-item{
  display: inline block;
  color: white;
  text-decoration: none;
  text-align: center;
  margin: 25px
}

nav{
  height: 80px;
}

#hamburger{
  display: none;
}

@media(max-width: 861px){
  .menu-item-container{
    display: none; 
    background-color: #282c34;
  }
  .menu-item{
    margin: 25px 15px;
  }
  #hamburger{
    display: inline-block;
    width: 40px;
    padding: 6px 40px
  }
}

