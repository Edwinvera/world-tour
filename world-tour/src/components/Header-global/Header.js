import React, {useState} from 'react'
import {Link} from 'react-router-dom'
import earth from '../../images/Emblem-earth.svg'
import hamburger from '../../images/hamburger.png'
import './Header.css'
// import styled from 'styled-components'



function Header() {

  const [open, updateOpen] = useState(false) 


  return (
    <div>

      <header className="app-header">
        <nav>
          <ul>
            {/* <li id= "hamburger-container"> DONT NEED*/}
            <img id="hamburger" src={hamburger} alt="hamburger menu" onClick={(e) => updateOpen(!open)}/>
            {/* {console.log(open)} */}
            {/* </li> DONT NEED*/}
            <li className= "menu-item-container" style={open ? { display:"block" }: {display:""}}>
              <Link to='/' className="menu-item" style={open ? { display:"block" }: {display:""}}>
                  Home
              </Link>
            </li>
            <li className= "menu-item-container" style={open ? { display:"block" }: {display:""}}>
              <Link to='/about' className="menu-item" style={open ? { display:"block" }: {display:""}}>
                  About
              </Link>
            </li>
            <li className= "menu-item-container" style={open ? { display:"block" }: {display:""}}>
              <Link to='/contact' className="menu-item" style={open ? { display:"block" }: {display:""}}> 
                  Contact
              </Link>
            </li>
          </ul>
        </nav>
        <div>
          <Link to='/explorer'>
            <img src={earth} className="earth-logo" alt="logo" />
          </Link>
        </div>
        
      </header>
    </div>
  )
}

export default Header
