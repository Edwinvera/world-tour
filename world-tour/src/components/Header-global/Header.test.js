import renderer from 'react-test-renderer'
import 'jest-styled-components'
import { mount } from "enzyme"
import React from 'react'
import Header, { Earth, Hamburger, AppHeader}from './Header'

test('Header_Earth component renders', () => {
  const EarthComponent = renderer.create(<Earth/>).toJSON()
  expect(EarthComponent).toMatchSnapshot()
})

test('Header_Hamburger component renders', () => {
  const HamburgerComponent = renderer.create(<Hamburger/>).toJSON()
  expect(HamburgerComponent).toMatchSnapshot()
})

test('Header_AppHeader component renders', () => {
  const AppHeaderComponent = renderer.create(<AppHeader/>).toJSON()
  expect(AppHeaderComponent).toMatchSnapshot()
})
