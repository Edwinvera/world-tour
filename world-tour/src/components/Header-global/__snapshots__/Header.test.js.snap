// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header_AppHeader component renders 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  background-color: #282c34;
  opacity: .85;
}

<header
  className="c0"
/>
`;

exports[`Header_Earth component renders 1`] = `
.c0 {
  position: absolute;
  height: 170px;
  border-radius: 360px;
  box-shadow: 5px 10px 20px #F4D35E;
  cursor: pointer;
  -webkit-animation: fvtopB infinite 50s linear;
  animation: fvtopB infinite 50s linear;
}

.c0:hover {
  box-shadow: 5px 10px 40px #F4D35E;
}

@media (max-width:861px) {
  .c0 {
    height: 150px;
    right: 41vw;
  }
}

@media (max-width:730px) {
  .c0 {
    right: 39vw;
  }
}

@media (max-width:650px) {
  .c0 {
    right: 37vw;
  }
}

@media (max-width:500px) {
  .c0 {
    right: 35vw;
  }
}

@media (max-width:470px) {
  .c0 {
    right: 32vw;
  }
}

@media (max-width:415px) {
  .c0 {
    right: 30vw;
  }
}

@media (max-width:365px) {
  .c0 {
    top: 22vw;
    right: 30vw;
    box-shadow: 5px 10px 30px #F4D35E;
  }
}

<img
  className="c0"
/>
`;

exports[`Header_Hamburger component renders 1`] = `
.c0 {
  display: none;
}

@media (max-width:861px) {
  .c0 {
    display: inline-block;
    width: 40px;
    padding: 0 40px;
  }
}

@media (max-width:365px) {
  .c0 {
    width: 15px;
  }
}

<img
  className="c0"
/>
`;
